src/main/java/com/test/KafkaTestApp/config/Config.java

```diff
@@ -0,0 +1,42 @@
+package com.test.KafkaTestApp.config;
+
+import static com.test.KafkaTestApp.client.ClientManager.CLIENT_KAFKA_TOPIC;
+
+import com.test.KafkaTestApp.client.Client;
+import org.springframework.boot.autoconfigure.kafka.KafkaProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.kafka.config.TopicBuilder;
+import org.springframework.kafka.core.ConsumerFactory;
+import org.springframework.kafka.core.DefaultKafkaConsumerFactory;
+import org.springframework.kafka.core.DefaultKafkaProducerFactory;
+import org.springframework.kafka.core.KafkaAdmin.NewTopics;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.core.ProducerFactory;
+
+@Configuration
+public class Config {
+
+  @Bean
+  public ConsumerFactory<String, Client> consumerFactory(KafkaProperties properties) {
+    return new DefaultKafkaConsumerFactory<>(properties.buildConsumerProperties());
+  }
+
+  @Bean
+  public ProducerFactory<String, Client> clientProducerFactory(KafkaProperties properties) {
+    return new DefaultKafkaProducerFactory<>(properties.buildProducerProperties());
+  }
+
+  @Bean
+  public KafkaTemplate<String, Client> clientTemplate(ProducerFactory<String, Client> factory) {
+    return new KafkaTemplate<>(factory);
+  }
+
+  @Bean
+  public NewTopics topics() {
```

Comment:
Better to no create topics separately not as part of the application. In our case you may create topics as part of the docker-compose file that could be responsible for for infrastructure setup (kafka + zookeeper / application);.